src/lib/atoms/performance/atom-family.ts

```
@@ -0,0 +1,45 @@
+// AtomFamily patterns for efficient dynamic atoms
+import { atom } from 'jotai'
+import { atomFamily } from 'jotai/utils'
+
+import type { Workout, TrainingPlan } from '@/lib/supabase'
+
+// Workout atom family - creates individual atoms per workout ID
+export const workoutAtomFamily = atomFamily(
+  (workoutId: string) => atom<Workout | null>(null)
```

ESLint

[error] 'workoutId' is defined but never used. Allowed unused args must match /^_/u.

(@typescript-eslint/no-unused-vars)